postgres/src/main/java/org/jdbi/v3/postgres/PGObjectArgumentFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import org.jdbi.v3.core.argument.AbstractArgumentFactory;
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.postgresql.util.PGobject;
+
+/**
+ * Argument factory for {@link PGobject}.
+ */
+public class PGObjectArgumentFactory extends AbstractArgumentFactory<PGobject> {
+
+    public PGObjectArgumentFactory() {
+        super(Types.OTHER);
+    }
+
+    @Override
+    protected Argument build(PGobject value, ConfigRegistry config) {
+        return new Argument() {
+
+            @Override
+            public void apply(int position, PreparedStatement statement, StatementContext ctx) throws SQLException {
+                ((org.postgresql.PGConnection) statement.getConnection()).addDataType(PostgresTypes.getTypeName(value.getClass()), value.getClass());
```

Comment:
Personal nitpick: I'd like to see the cast object extracted to a variable (`PGConnection conn = ... `), because this line is pretty long and hard to read, and I think there needs to be a `SuppressWarnings` on the cast, so extracting it to a variable also neatly limits the scope of the suppression.